plugins/processors/parser/parser.go

```diff
@@ -16,17 +17,17 @@ type Parser struct {
 	parser       telegraf.Parser
 }
 
-func (p *Parser) Apply(metrics ...telegraf.Metric) []telegraf.Metric {
-	if p.parser == nil {
-		var err error
-		p.parser, err = parsers.NewParser(&p.Config)
-		if err != nil {
-			p.Log.Errorf("could not create parser: %v", err)
-			return metrics
-		}
-		models.SetLoggerOnPlugin(p.parser, p.Log)
+func (p *Parser) Init() (err error) {
+	// Initialise parser
+	p.parser, err = parsers.NewParser(&p.Config)
+	if err != nil {
+		return fmt.Errorf("could not create parser: %v", err)
 	}
 
+	return err
+}
```

Comment:
```suggestion
	return nil
}
```

Reply:
That's exactly the same right? Why the preference for the other?

Reply:
Because it is explicit. Everybody directly sees that the function is successful here without backtracing the value of `err`.